packages/webpack/src/config/base.js

```diff
@@ -184,14 +193,31 @@ export default class WebpackBaseConfig {
     // Prioritize nested node_modules in webpack search path (#2558)
     const webpackModulesDir = ['node_modules'].concat(this.buildContext.options.modulesDir)
 
+    const {
+      options: {
+        buildDir,
+        build: {
+          template = '@nuxt/vue-app'
+        }
+      }
+    } = this.buildContext
+
     return {
       resolve: {
         extensions: ['.wasm', '.mjs', '.js', '.json', '.vue', '.jsx'],
         alias: this.alias(),
-        modules: webpackModulesDir
+        modules: webpackModulesDir,
+        plugins: [
+          PnpWebpackPlugin.bind(buildDir, module, '@nuxt/babel-preset-app'),
```

Comment:
(just out of curiosity) is there any way that we detect Nuxt is running via PnP and conditionally add plugin?

Reply:
The plugin already works this way, [it's a noop unless PnP is detected](https://github.com/arcanis/pnp-webpack-plugin/blob/master/index.js#L127-L131) 🙂

Reply:
Awesome thanks for clearance @arcanis!